numpy/core/src/umath/matmul.c.src

```diff
@@ -79,24 +81,30 @@ NPY_NO_EXPORT void
     enum CBLAS_ORDER order = CblasRowMajor;
     enum CBLAS_TRANSPOSE trans1, trans2;
     int M, N, P, lda, ldb;
-    M = m;
-    N = n;
-    P = p;
+    assert(m < NPY_MAX_INT);
+    assert(n < NPY_MAX_INT);
+    assert(p < NPY_MAX_INT);
+    M = (int)m;
+    N = (int)n;
+    P = (int)p;
 
     if (is1_m == sizeof(@type@)) {
         trans1 = CblasTrans;
-        lda = N > 1 ? is1_n / sizeof(@type@) : 1;
+        assert((N > 1) && (is1_n / sizeof(@type@) < NPY_MAX_INT));
+        lda = N > 1 ? (int)(is1_n / sizeof(@type@)) : 1;
     }
     else {
         /* If not ColMajor, caller should have ensured we are RowMajor */
         /* will not assert in release mode */
         assert(is1_n == sizeof(@type@));
         trans1 = CblasNoTrans;
-        lda = N > 1 ? is1_m / sizeof(@type@) : 1;
+        assert((N > 1) && (is1_m / sizeof(@type@) < NPY_MAX_INT));
+        lda = N > 1 ? (int)(is1_m / sizeof(@type@)) : 1;
     }
     if (is2_n == sizeof(@type@)) {
         trans2 = CblasTrans;
-        ldb = N > 1 ? is2_p / sizeof(@type@) : 1;
+        assert((N > 1) && (is2_p / sizeof(@type@) < NPY_MAX_INT));
+        ldb = N > 1 ? (int)(is2_p / sizeof(@type@)) : 1;
```

Comment:
Does cblas have the same requirements as lapack that `ldb > 1`? Will this fail if `is2_p == 0`?

Reply:
maybe, see #9165. We will not encounter that now since we demand contiguous input and output